frontend/context/StakingProgramProvider.tsx

```diff
@@ -1,73 +1,88 @@
+export const INITIAL_DEFAULT_STAKING_PROGRAM_ID = StakingProgramId.PearlBeta;
 
 export const StakingProgramContext = createContext<{
   activeStakingProgramId?: StakingProgramId | null;
   defaultStakingProgramId: StakingProgramId;
-  updateActiveStakingProgramId: () => Promise<void>;
-  setDefaultStakingProgramId: (stakingProgramId: StakingProgramId) => void;
 }>({
   activeStakingProgramId: undefined,
   defaultStakingProgramId: INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  updateActiveStakingProgramId: async () => {},
-  setDefaultStakingProgramId: () => {},
 });
 
-/** Determines the current active staking program, if any */
-export const StakingProgramProvider = ({ children }: PropsWithChildren) => {
-  const { service } = useServices();
+const currentAgent = AGENT_CONFIG.trader; // TODO: replace with dynamic agent selection
+const currentChainId = GNOSIS_CHAIN_CONFIG.chainId; // TODO: replace with dynamic chain selection
 
-  const [activeStakingProgramId, setActiveStakingProgramId] =
-    useState<StakingProgramId | null>();
+/**
+ * hook to get the active staking program id
+ */
+const useGetActiveStakingProgramId = () => {
+  const queryClient = useQueryClient();
 
-  const [defaultStakingProgramId, setDefaultStakingProgramId] = useState(
-    INITIAL_DEFAULT_STAKING_PROGRAM_ID,
-  );
+  const { selectedService, isFetched: isLoaded } = useServices();
+  const serviceConfigId =
+    isLoaded && selectedService ? selectedService?.service_config_id : '';
+  const { service } = useService({ serviceConfigId });
 
-  const updateActiveStakingProgramId = useCallback(async () => {
-    // if no service nft, not staked
-    const serviceId =
-      service?.chain_configs[CHAIN_CONFIG.OPTIMISM.chainId].chain_data?.token;
+  // if no service nft, not staked
+  const serviceId = service?.chain_configs[currentChainId].chain_data?.token;
 
-    if (
-      !service?.chain_configs[CHAIN_CONFIG.OPTIMISM.chainId].chain_data?.token
-    ) {
-      setActiveStakingProgramId(null);
-      return;
-    }
+  const response = useQuery({
+    queryKey: [REACT_QUERY_KEYS.STAKING_PROGRAM_KEY, currentChainId, serviceId],
+    queryFn: async () => {
+      return await currentAgent.serviceApi.getCurrentStakingProgramByServiceId(
+        serviceId!,
+        currentChainId,
+      );
+      // if the response is null, return default staking program id. Thoughts?
+    },
+    enabled: !!serviceId,
+    refetchInterval: isLoaded ? FIVE_SECONDS_INTERVAL : false,
+  });
 
-    if (serviceId) {
-      // if service exists, we need to check if it is staked
-      AutonolasService.getCurrentStakingProgramByServiceId(serviceId).then(
-        (stakingProgramId) => {
-          setActiveStakingProgramId(stakingProgramId);
-        },
+  const setActiveStakingProgramId = useCallback(
+    (stakingProgramId: StakingProgramId | null) => {
+      queryClient.setQueryData(
+        [REACT_QUERY_KEYS.STAKING_PROGRAM_KEY, currentChainId, serviceId],
+        stakingProgramId,
```

Comment:
a bit confused by this, i think this function should accept `chainId and serviceId` as params, instead of relying solely on values in context